app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    def wrapper(*args: Any) -> Any:
+        if args in func_cache:
+            print("Getting from cache")
+            return func_cache[args]
+        else:
+            result = func(*args)
+            print("Calculating new result")
+            func_cache[args] = result
+            return result
```

Comment:
```suggestion
            func_cache[args] = func(*args)
            print("Calculating new result")            
            return func_cache[args]
```
Code can be more simplify if you dont use one more variable.